### DISCUSSION TITLE
4th (Rotary) axis for mill on ESP32

### DISCUSSION THREAD

**Discussion by imechura** [grblHAL/core#552] [General]
Is there a guide that shows how to enable a 4th axis on grblhal? If not can someone point me in the right direction? I am running on ESP32 the 4th axis is i'm trying to use is a rotary indexer.

**Comment by terjeio** [comment]
> Is there a guide that shows how to enable a 4th axis on grblhal?

I have not made any.

Number of axes is [defined here](https://github.com/grblHAL/core/blob/8b9b7859ee6b370442d380d3d754fa5392b064b2/config.h#L37-L43) if you are compiling locally, if using the [Web Builder](http://svn.io-engineering.com:8080/?driver=ESP32) with the _Number of axes_ drop down in the _General_ tab.

`$376` is used to specify axes > 3 as rotary, send `$$=376` for info. `$538` is a [brand new setting](https://github.com/grblHAL/core/blob/master/changelog.md#20240719) that allows fast 'rewind' of rotary axes.
